Update SDK to 8.0.100-rc.1.23381.2

eerhardt opened the pull request:
Testing for https://github.com/dotnet/runtime/issues/89734. Trying to see if we can repro the issue.

eerhardt committed:
Update SDK to 8.0.100-rc.1.23381.2

eerhardt commented:
@lewing @radical @javiercn - Looks like everything is failing with 

```
/Users/runner/work/1/s/.dotnet/sdk/8.0.100-rc.1.23381.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: wasm-tools [/Users/runner/work/1/s/src/Components/WebAssembly/testassets/WasmLinkerTest/WasmLinkerTest.csproj]
/Users/runner/work/1/s/.dotnet/sdk/8.0.100-rc.1.23381.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload restore [/Users/runner/work/1/s/src/Components/WebAssembly/testassets/WasmLinkerTest/WasmLinkerTest.csproj]
```

Can someone fix it?

javiercn commented:
@eerhardt did this fail on retry?

lewing commented:
@radical can you take a look?

eerhardt committed:
Respond to new code analysis warnings.

eerhardt committed:
Fix up previous changes

BrennanConroy commented on src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs:
```diff
@@ -26,7 +26,7 @@ public static CommandOption Option(this CommandLineApplication command, string t
             template,
             description,
-            template.IndexOf("<", StringComparison.Ordinal) != -1
+            template.Contains("<")
```
```suggestion
            template.Contains('<')
```

eerhardt replied:
This just re-broke the build... :( I'm reverting.

BrennanConroy replied:
Reverting won't fix it either. This must be used in multiple tfms, one without char based `Contains` method.

eerhardt replied:
Reverting your change will fix it.

eerhardt replied:
This set of analysis rules is terrible. Am I really supposed to #if here?

eerhardt replied:
This is a bad experience.

radical committed:
WasmLinkerTest: Workaround incorrect settings in the project file

Issue when updating to build with rc1:
```
/Users/runner/work/1/s/.dotnet/sdk/8.0.100-rc.1.23381.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To build this project, the following workloads must be installed: wasm-tools [/Users/runner/work/1/s/src/Components/WebAssembly/testassets/WasmLinkerTest/WasmLinkerTest.csproj]
/Users/runner/work/1/s/.dotnet/sdk/8.0.100-rc.1.23381.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.ImportWorkloads.targets(38,5): error NETSDK1147: To install these workloads, run the following command: dotnet workload restore [/Users/runner/work/1/s/src/Components/WebAssembly/testassets/WasmLinkerTest/WasmLinkerTest.csproj]
```

Explanation:

1. This project has `RuntimeIdentifier=browser-wasm`, which triggers the
   workload manifests for wasm
2. But the project is *not* using blazor sdk, or the wasm-sdk.
3. The condition at https://github.com/dotnet/emsdk/blob/5466de081d350d3214c1ae68b7fca6295109d5cf/eng/nuget/Microsoft.NET.Workload.Emscripten.Current.Manifest/WorkloadManifest.targets#L12

```xml
<UsingBrowserRuntimeWorkload Condition="'$(RunAOTCompilation)' == 'true' or '$(UsingMicrosoftNETSdkBlazorWebAssembly)' != 'true' or '$(UsingMicrosoftNETSdkWebAssembly)' != 'true'" >true</UsingBrowserRuntimeWorkload>
```

   .. means that if a `rid=browser-wasm` project is not using blazor sdk,
   or the wasm sdk, then it depends on the targets in the workload. And
   hence requires the workload.

Solution:
As a workaround set `UsingMicrosoftNETSdkWebAssembly=true` also
(`UsingMicrosoftNETSdkBlazorWebAssembly` is already set to true).

BrennanConroy committed:
Update src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs

eerhardt committed:
Revert "Update src/Tools/Shared/CommandLine/CommandLineApplicationExtensions.cs"

This reverts commit e36a69222e4b38c33c4bc17c53413e9c15a89ffc.

eerhardt committed:
Fix build on non-.NET Core.

eerhardt committed:
More fixes.

eerhardt committed:
Fix ViewPath
